src/RedHat.AspNetCore.Server.Kestrel.Transport.Linux/TransportThread.TSocket.cs

```diff
@@ -116,13 +135,31 @@ public int PendingEventState
                 set => _flags = (SocketFlags)value;
             }
 
+            public override void Abort(ConnectionAbortedException abortReason)
+            {
+                CancelWriteToSocket();
+            }
+
             public override void Abort()
             {
                 CancelWriteToSocket();
             }
 
+            public override async ValueTask DisposeAsync()
+            {
+                Transport.Input.Complete();
+                Transport.Output.Complete();
+
+                Abort();
+
+                await _waitForConnectionClosedTcs.Task;
+                _connectionClosedTokenSource.Dispose();
+            }
+
             private void CancelWriteToSocket()
             {
+                Output.CancelPendingRead();
```

Comment:
Can you move `Output.CancelPendingRead()` into `Abort(ConnectionAbortedException)` and remove the `Abort()`?